src/sql/workbench/services/accountManagement/browser/accountManagementService.ts

```diff
@@ -363,8 +363,11 @@ export class AccountManagementService implements IAccountManagementService {
 	 * Copy the user code to the clipboard and open a browser to the verification URI
 	 */
 	public copyUserCodeAndOpenBrowser(userCode: string, uri: string): void {
-		this._clipboardService.writeText(userCode).catch(err => onUnexpectedError(err));
-		this._openerService.open(URI.parse(uri)).catch(err => onUnexpectedError(err));
+		this._clipboardService.writeText(userCode).then(() => {
```

Comment:
Even simpler I'd just make this function async and await the calls to writeText and open.

And FYI, not sure if this was on purpose but currently you're changing the behavior such that if the call to open fails it won't call onUnexpectedError. You'd need to either return the result of that or await it within the .then

Reply:
Wouldn't the entire call hierarchy require async and await keywords if I use them here?

No, that wasn't on purpose. I missed it when I added the curly braces. Return is implied when the lambda expression doesn't have curly braces, but that stops being the case when curly braces are used.

Reply:
You can have whatever is calling this choose not to be async itself (it could then just use .then/.catch as needed). And in this case you could just have handleOnAddAccount return a promise too since the callback doesn't care about the return value.